app/src/main/java/com/github/se/polyfit/viewmodel/dailyRecap/DailyRecapViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.github.se.polyfit.viewmodel.dailyRecap
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.se.polyfit.data.repository.MealRepository
+import com.github.se.polyfit.model.meal.Meal
+import dagger.hilt.android.lifecycle.HiltViewModel
+import java.time.LocalDate
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@HiltViewModel
+class DailyRecapViewModel @Inject constructor(private val mealRepository: MealRepository) :
+    ViewModel() {
+  private val _meals: MutableStateFlow<List<Meal>> = MutableStateFlow(mutableListOf())
+  val meals: StateFlow<List<Meal>> = _meals
+
+  private val _isFetching: MutableStateFlow<Boolean> = MutableStateFlow(false)
+  val isFetching: StateFlow<Boolean> = _isFetching
+
+  init {
+    getMealsOnDate(LocalDate.now())
+  }
```

Comment:
Be careful not with this line, if the viewmodel gets instantiated frequently it will result in multiple unnecessary fetching of the data. If this happens currently or later on an Idea would be to put a refresh mechanism on it and start fetching only when the user refreshes